File: HackerRank/kind.cpp
#include <bits/stdc++.h>

#define endl '\n'
#define pb push_back
#define mp make_pair


using namespace std;

int main()
{
	ios::sync_with_stdio(0);
	cin.tie(0);
	int n;
	cin >> n;
	vector<int> t;
	for(int i = 0; i < n; ++i)
	{
		int x;
		cin >> x;
		t.pb(x);
	}
	for(int i = 0; i < n; ++i)
		t.pb(t[i]);
	
	int ans = 0;
	int maxTot = 0;
	int tmp = 0;
	int minu = 0;
	// calculando resposta comecando do ultimo
	for(int k = n-1; k < t.size()-1; ++k)
	{
		if(t[k] <= minu)
			tmp++;
		minu++;	
	}
	maxTot = tmp;
	ans = n-1;
	for(int k = n-2; k >= 0; --k)
	{
		cout << t[k] << " o tk da vez eh " << endl;
		int resp = tmp;
		if(t[k] == 0)
			resp++;
		if(t[k+n] <= n)
			resp--;
		cout << "o inicio eh o t[k] = " << t[k] << " e a resposta dele eh " << resp << endl;
		if(resp >= maxTot)
		{
			maxTot = resp;
			ans = k;
		}
                tmp = resp;
	}	
	cout << ans + 1 << endl;
	return 0;
	
}

File: HackerRank/kind.py
n = int(raw_input())
a = list(map(int, raw_input().split()))

ans = 0
maxTot = 0
tmp = 0
minu = 0

for x in range(n):
	minu = 0
	tmp = 0
	for k in range(x, n, 1):
		if a[k] <= minu:
			tmp = tmp + 1
		minu = minu + 1
	for j in range(0, x, 1):
		if a[j] <= minu:
			tmp = tmp + 1
		minu = minu + 1
	if tmp > maxTot:
		maxTot = tmp
		ans = x
		print(maxTot)
		print(x)
print(a)
print(ans + 1)
	

